test-utils/src/dev/test_cmds.rs

```diff
@@ -155,32 +155,24 @@ pub fn redact_variable(input: &str) -> String {
     .to_string();
 
     // Replace timestamps.
+    //
+    // Format: RFC 3339 (ISO 8601)
+    // Example: 1970-01-01T00:00:00Z
     let s = regex::Regex::new(r"\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}Z")
         .unwrap()
         .replace_all(&s, "<REDACTED_TIMESTAMP>")
         .to_string();
 
-    let s = {
-        let mut new_s = String::with_capacity(s.len());
-        let mut last_match = 0;
-        for m in regex::Regex::new(r"\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d+Z")
-            .unwrap()
-            .find_iter(&s)
-        {
-            new_s.push_str(&s[last_match..m.start()]);
-            new_s.push_str("<REDACTED");
-            // We know from our regex that `m.len()` is at least 2 greater than
-            // the length of "<REDACTEDTIMESTAMP>", so this subtraction can't
-            // underflow. Insert spaces to match widths.
-            for _ in 0..(m.len() - "<REDACTEDTIMESTAMP>".len()) {
-                new_s.push(' ');
-            }
-            new_s.push_str("TIMESTAMP>");
-            last_match = m.end();
-        }
-        new_s.push_str(&s[last_match..]);
-        new_s
-    };
+    // Replace timestamps that have a sub-second portion.
+    // Example: 1970-01-01T00:00:00.000001Z
+    //
+    // Note that depending on the amount of trailing zeros,
+    // this value can have different widths. However, "<REDACTED_TIMESTAMP>"
+    // has a deterministic width, so that's used instead.
+    let s = regex::Regex::new(r"\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d+Z")
+        .unwrap()
+        .replace_all(&s, "<REDACTED_TIMESTAMP>")
+        .to_string();
```

Comment:
GitHub won't let me write a suggestion here but I suspect combining these into a single regex would work? e.g. `r"\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}(\.\d+)?Z"`

Reply:
That seems valid -- updated!